src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.StringSplitter.cs

```diff
@@ -46,6 +44,27 @@ private abstract class StringSplitter
                 TabSize = tabSize;
                 _indentStyle = indentStyle;
                 CancellationToken = cancellationToken;
+
+                var options = document.GetOptionsAsync().Result;
+                var newLineKey = new OptionKey(NewLine, LanguageNames.CSharp);
+                var newLineValue = (string)options.GetOption(newLineKey);
+
+                // Check if there is a line ending specified in the editorconfig
+                // and ensure we honour it. 
+                if (!string.IsNullOrEmpty(newLineValue) && newLineValue == "\n")
+                {
+                    PlusNewLineToken = SyntaxFactory.Token(
+                   leading: default,
+                   SyntaxKind.PlusToken,
+                   SyntaxFactory.TriviaList(SyntaxFactory.ElasticLineFeed));
+                }
+                else
+                {
+                    PlusNewLineToken = SyntaxFactory.Token(
+                    leading: default,
+                    SyntaxKind.PlusToken,
+                    SyntaxFactory.TriviaList(SyntaxFactory.ElasticCarriageReturnLineFeed));
+                }
```

Comment:
Can simplify all of this to just make the token but only conditionally switch the type of elastic trivia to add

Reply:
Done